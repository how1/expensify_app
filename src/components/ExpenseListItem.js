import React from 'react';
import { NavLink } from 'react-router-dom';

const ExpenseListItem = ({id, description, amount, createdAt}) => {
	return (
		<div>
			<NavLink 
				activeClassName="is-active" 
				exact={true} 
				to={"/edit/" + id}
			>
			<h3>{description}</h3>
			</NavLink>
			<p>{amount} | {createdAt}</p>
		</div>
)};



export default ExpenseListItem;