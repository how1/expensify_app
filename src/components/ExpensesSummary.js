import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import { connect } from 'react-redux';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {
	render() {
		const formattedExpensesTotal = numeral(this.props.expensesTotal / 100).format('$0,0.00');
		return (
			<div>
				<h1>Viewing {this.props.expenseCount} expense(s) totalling {formattedExpensesTotal}</h1>
			</div>
		);
	};
};

const mapStateToProps = (state) => {
	const expenses = selectExpenses(state.expenses, state.filters);
	return {
		expenses,
		expenseCount: expenses.length,
		expensesTotal: selectExpensesTotal(expenses)
	};
};

export default connect(mapStateToProps)(ExpensesSummary);