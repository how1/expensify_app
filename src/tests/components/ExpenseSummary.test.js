import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should handle summary for 0 expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenses={[]}/>);
	expect(wrapper).toMatchSnapshot();
});

test('should handle summary for 1 expense', () => {
	const wrapper = shallow(<ExpensesSummary expenses={[expenses[1]]}/>);
	expect(wrapper).toMatchSnapshot();
});

test('should handle summary for multiple expenses', () => {
	const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
	expect(wrapper).toMatchSnapshot();
});