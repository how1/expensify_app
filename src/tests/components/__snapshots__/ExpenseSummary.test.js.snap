// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle summary for 0 expenses 1`] = `
<div>
  <h1>
    Viewing 
     expense(s) totalling 
    $0.00
  </h1>
</div>
`;

exports[`should handle summary for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
     expense(s) totalling 
    $0.00
  </h1>
</div>
`;

exports[`should handle summary for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
     expense(s) totalling 
    $0.00
  </h1>
</div>
`;
